### ISSUE TITLE
Question about blip2

### ISSUE THREAD

**Issue by shoutOutYangJie** [salesforce/LAVIS#360]
when computing "Image Captioning" loss, I find some weird problem.

'''
      decoder_input_ids = text_tokens.input_ids.clone()
        decoder_input_ids[:, 0] = self.tokenizer.bos_token_id
        labels = decoder_input_ids.masked_fill(
            decoder_input_ids == self.tokenizer.pad_token_id, -100
        )

        query_atts = torch.ones(query_tokens.size()[:-1], dtype=torch.long).to(
            image.device
        )
        attention_mask = torch.cat([query_atts, text_tokens.attention_mask], dim=1)
        lm_output = self.Qformer(
            decoder_input_ids,
            attention_mask=attention_mask,
            past_key_values=query_output.past_key_values,
            return_dict=True,
            labels=labels,
        )
'''
You send all text token to Qfromer, which uses self-attention.  So the network has seen all tokens. but you still try use caption loss by next prediction. It is non-sense

**Comment by LiJunnan1992** [comment]
This is the standard teacher-forcing method to compute language modeling loss.

**Comment by shoutOutYangJie** [comment]
> This is the standard teacher-forcing method to compute language modeling loss.

Could you share the paper which utilizes this loss ? 
I think if network sees all tokens and predicts next word, it is meanless, because the network alreay know what the ground truth is.
